orb-connd/src/service/mod.rs

```diff
@@ -0,0 +1,481 @@
+                    )))
+                }
+            }
+
+            info!("applied wifi qr successfully");
+
+            Ok(())
+        }
+        .await
+        .inspect_err(|e| error!("failed to apply wifi qr with {e}"))
+    }
+
+    async fn apply_netconfig_qr(
+        &self,
+        contents: String,
+        check_ts: bool,
+    ) -> ZResult<()> {
+        async {
+            info!("trying to apply netconfig qr code");
+            NetConfig::verify_signature(&contents, self.release).into_z()?;
+            let netconf = NetConfig::parse(&contents).into_z()?;
+
+            if check_ts {
+                let now = Utc::now();
+                let delta = now - netconf.created_at;
+                if delta.num_minutes() > 10 {
+                    return Err(e("qr code was created more than 10min ago"));
+                }
+            }
+
+            if let Some(wifi_creds) = netconf.wifi_credentials {
+                let sec: WifiSec = wifi_creds.auth.try_into().into_z()?;
+                info!(ssid = wifi_creds.ssid, "adding wifi network from netconfig");
+
+                let saved_profile = self
+                    .nm
+                    .list_wifi_profiles()
+                    .await
+                    .into_z()?
+                    .into_iter()
+                    .find(|p| p.ssid == wifi_creds.ssid);
+
+                match (saved_profile, wifi_creds.psk) {
+                    // profile exists and no pwd was provided
+                    (Some(profile), None) => {
+                        self.nm
+                            .connect_to_wifi(&profile.path, Self::DEFAULT_WIFI_IFACE)
+                            .await
+                            .into_z()?;
+                    }
+
+                    // pwd was provided so we assume a new profile is being added
+                    (Some(_), Some(psk)) | (None, Some(psk)) => {
+                        self.add_wifi_profile(
+                            wifi_creds.ssid,
+                            sec.as_str().to_string(),
+                            psk.0,
+                            wifi_creds.hidden,
+                        )
+                        .await?;
+                    }
+
+                    // no pwd provided and no existing profile, nothing we can do
+                    (None, None) => {
+                        return Err(e(&format!(
+                        "wifi profile '{}' does not exist and no password was provided",
+                        wifi_creds.ssid,
+                    )))
+                    }
+                }
+            };
+
+            // Pearl orbs do not support extra NetConfig fields
+            if self.platform == OrbOsPlatform::Pearl {
+                return Ok(());
+            }
+
+            if let Some(_airplane_mode) = netconf.airplane_mode {
+                warn!("airplane mode is not supported yet!");
+            }
+
+            if let Some(wifi_enabled) = netconf.wifi_enabled {
```

Comment:
If the orb is wifi-only, do we want to disable wifi via netconfig ? maybe only apply if it's lte ?